src/Nox.Reference.VatNumbers/Services/BrazilValidationService.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Nox.Reference.VatNumbers.Constants;
+using Nox.Reference.VatNumbers.Extension;
+using Nox.Reference.VatNumbers.Models;
+
+namespace Nox.Reference.VatNumbers.Services
+{
+    // Verifiend manually: FALSE
+    // Personal data cleaned: TRUE
+    internal class BrazilValidationService : IVatValidationService
+    {
+        // TODO: review
+        private const string _validationPattern = @"^\d{14}$";
+        private const string _validationPatternDescription = "VAT should have 14 numeric characters";
+
+        public ValidationResult ValidateVatNumber(IVatNumber vatNumber)
+        {
+            // Cannot have special characters
+            // Can have or not have 'BR' prefix
+            var number = vatNumber.NormalizeVatNumber();
+
+            if (string.IsNullOrWhiteSpace(number))
+            {
+                return ValidationResults.NullValidationResult;
+            }
+
+            var result = new ValidationResult();
+
+            // Code should match the pattern
+            IVatValidationService.ValidateRegex(result, number, _validationPattern, vatNumber.Number, _validationPatternDescription);
+
+            // Should be consisting of numbers to check checksum
+            number.ValidateCustomChecksum(result, CalculateChecksum);
+
+            return result;
+        }
+
+        private static bool CalculateChecksum(string number, ValidationResult result)
+        {
+            var minimumLengthRequirement = 14;
+            if (number.Length < minimumLengthRequirement)
+            {
+                result.ValidationErrors.Add(string.Format(ValidationErrors.MinimumNumbericLengthError, minimumLengthRequirement));
+                return false;
+            }
+
+            var registration = number.Substring(0, 12);
+            registration += DigitChecksum(registration);
```

Comment:
Can we introduce loop here. Is  2 digit in registration.Substring(registration.Length - 2)  dependent on loop iteractions?

Reply:
No, should be fine like this